compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -828,7 +828,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             LifetimeRes::Fresh { param, kind, .. } => {
                 // Late resolution delegates to us the creation of the `LocalDefId`.
                 let _def_id = self.create_def(
-                    self.current_hir_id_owner.def_id, // FIXME: should this use self.current_def_id_parent?
+                    self.current_def_id_parent,
```

Comment:
Should we remove that parameter and have `create_def` always use `current_def_id_parent`?

Reply:
That would make sense to me :+1: I think using `current_hir_id_owner` only makes sense if either:
- We statically know that the `current_def_id_parent` is the innermost hir owner
- All definitions participating in the def parenting tree are also hir owners (aka, the former point is always trivially true)

I think right now there should never be any cases where we're creating defs that wants to skip a parent. The struct definition in `fn foo() -> Foo<{ struct Bar; 1 }>` *does* have that behaviour since its parent is `foo` not `constant#0` but that' happens via creating the `DefId` in `DefCollector`.